lib/atom-css-sniffer.js

```diff
@@ -4,16 +4,96 @@ let helpers;
 let path;
 let filesystem;
 
+// Caches
+let versions = {};
+let cmds = {}
+
 function loadDeps() {
-  if (!helpers) {
-    helpers = require('atom-linter');
-  }
-  if (!path) {
-    path = require('path');
-  }
-  if (!filesystem) {
-    filesystem = require('fs');
-  }
+    if (!helpers) {
+        helpers = require('atom-linter');
+    }
+    if (!path) {
+        path = require('path');
+    }
+    if (!filesystem) {
+        filesystem = require('fs');
+    }
+}
+
+function _getCommand(projectPath) {
+    if (cmds[projectPath]) {
+        return cmds[projectPath];
+    }
+
+    let cmd = projectPath + '/vendor/bin/css-sniff';
+
+    if (!filesystem.existsSync(cmd)) {
+        // do we have a WWW folder?
+        cmd = projectPath + '/www/vendor/bin/css-sniff';
+
+        if (!filesystem.existsSync(cmd)) {
+            cmd = null;
+        }
+    }
+
+    return cmds[projectPath] = cmd;
+}
+
+function _checkVersion(projectPath, cmd) {
+    if (versions[projectPath]) {
+        return versions[projectPath];
+    }
+
+    return versions[projectPath] = helpers.exec(cmd, ['--version'], {}).then(function (result) {
+        let matches = result.match(/v([0-9]\.[0-9])/);
+
+        return matches === null ? '1.0' : matches[1];
+    })
+}
+
+function _processResultV2(data, editorPath) {
+    let errors = [];
+
+    for (let i in data.files) {
```

Comment:
I think our JS codestyle defines that we should define all variables of the same scope in one define block, i.e.:

```JS
let errors = [],
    i;
```

Reply:
I'm not sure if I agree that this is more readable...

```js
let errors = [], i;

for (i = 0; i < data.length; i++) {
    // ...
}
```